src/multi_vector_simulator/E3_indicator_calculation.py

```diff
@@ -594,29 +643,12 @@ def add_onsite_energy_fraction(dict_values):
     - test_equation_onsite_energy_fraction()
     """
 
-    total_generation = (
-        dict_values[KPI][KPI_SCALARS_DICT][TOTAL_RENEWABLE_GENERATION_IN_LES]
-        + dict_values[KPI][KPI_SCALARS_DICT][TOTAL_NON_RENEWABLE_GENERATION_IN_LES]
-    )
+    total_generation = dict_values[KPI][KPI_SCALARS_DICT][TOTAL_GENERATION_IN_LES]
 
-    total_feedin = 0
-    # sum up all feed into the grid
-    for dso in dict_values[ENERGY_PROVIDERS]:
-        # Get source connected to the specific DSO in question
-        scalars_matrix = dict_values[KPI][KPI_SCALAR_MATRIX]
-        # load total flow into the dso sink
-        dso_sink = scalars_matrix.loc[
-            scalars_matrix["label"]
-            == dict_values[ENERGY_PROVIDERS][  # todo: IS this referece always correct??
-                dso
-            ][CONNECTED_FEEDIN_SINK]
-            + AUTO_SINK
-        ]
-        dso_feedin = dso_sink["total_flow"][0]
-        # sum up all dso flows
-        total_feedin = total_feedin + dso_feedin
-    # save total feedin into KPI_SCALARS_DICT
-    dict_values[KPI][KPI_SCALARS_DICT].update({TOTAL_FEEDIN: total_feedin})
+    add_total_feedin_electricity_equivaluent(dict_values)
```

Comment:
I would propably call this from `E0`, so that one only has to look at one code section to find the function calls for each of the technical KPI.

Reply:
ok!